webapp/graphite/metrics/views.py

```diff
@@ -290,6 +295,19 @@ def tree_json(nodes, base_path, wildcards=False):
   return results
 
 
+def nodes_by_position(matches, position):
+  found = set()
+
+  for metric in matches:
+    nodes = metric.path.split('.')
+    node = nodes[position]
+    if node in found:
+      continue
```

Comment:
No need for membership test, simply `.add()` and the set takes care of duplicates